pkg/experiment/metastore/dlq/recovery.go

```diff
@@ -0,0 +1,152 @@
+package dlq
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	metastorev1 "github.com/grafana/pyroscope/api/gen/proto/go/metastore/v1"
+	"github.com/grafana/pyroscope/pkg/tenant"
+
+	objstore "github.com/thanos-io/objstore"
+	"io"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+type RecoveryConfig struct {
+	Period time.Duration
+}
+
+type LocalServer interface {
+	AddBlock(context.Context, *metastorev1.AddBlockRequest) (*metastorev1.AddBlockResponse, error)
+}
```

Comment:
I think it may make sense to distinguish blocks added directly and through the recovery mechanism (e.g.: distinct metrics, online stream statistics, etc.)

```go
type LocalMetastore interface {
    AddRecoveredBlock(context.Context, *metastorev1.AddBlockRequest) (*metastorev1.AddBlockResponse, error)
}
```

Reply:
:+1: changed